docs/dev/plugins/other.rst

```diff
@@ -0,0 +1,391 @@
+Further Reading
+===============
+
+.. contents:: Table of Contents
+    :local:
+    :depth: 2
+
+Extending the Autotagger
+------------------------
+
+.. currentmodule:: beets.metadata_plugins
+
+Plugins can also be used to extend the autotagger functions i.e. the metadata
+lookup from external sources. For this your plugin has to extend the
+:py:class:`MetadataSourcePlugin` base class and implement all abstract methods.
+
+On metadata lookup, the autotagger will try to find matching candidates from all
+enabled metadata source plugins. To do this, we will call the
+:py:meth:`MetadataSourcePlugin.candidates` (or
+:py:meth:`MetadataSourcePlugin.item_candidates`) with all available (local)
+metadata. The list of retrieved candidates will be ranked by their
+:py:meth:`MetadataSourcePlugin.album_distance` (or
+:py:meth:`MetadataSourcePlugin.track_distance`) and be presented to the user for
+selection (or automatically selected if the threshold is met).
```

Comment:
A fair bit of detail has disappeared from this paragraph. If we don't want to expand about each of the methods in detail here, we should probably add the removed documentation to their docstrings and refer to them accordingly.

I think we should mention `album_for_id` and `track_for_id` methods instead of `album_distance` and `traack_distance` since the former are abstract while the latter aren't.